Lib/test/test_asyncio/test_timeouts.py

```diff
@@ -303,6 +351,39 @@ async def test_timeout_without_task(self):
         with self.assertRaisesRegex(RuntimeError, "has not been entered"):
             cm.reschedule(0.02)
 
+    async def test_timeout_taskgroup(self):
+        async def task():
+            try:
+                await asyncio.sleep(2)  # Will be interrupted after 0.01 second
+            finally:
+                1/0  # Crash in cleanup
+
+        with self.assertRaises(ExceptionGroup) as cm:
+            async with asyncio.timeout(0.01):
+                async with asyncio.TaskGroup() as tg:
+                    tg.create_task(task())
+                    try:
+                        raise ValueError
+                    finally:
+                        await asyncio.sleep(1)
+        eg = cm.exception
```

Comment:
Here, too, a comment might clarify what we're expecting. (I vaguely recall that @iritkatriel wrote some powerful helpers to check expected exception groups -- that might be overkill here, but they were useful to be able to follow the code.)